### ISSUE TITLE
2.4.0 Causes Fatal Error In Lumen

### ISSUE THREAD

**Issue by freshleafmedia** [getsentry/sentry-laravel#461]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
sentry/sentry-laravel:2.4.0

PHP: 8.0.2 using `declare(strict_types=1)`

### Steps to Reproduce

1. Install latest version of Lumen framework (8.2.3 at time of writing) and sentry/sentry-laravel (2.4.0) via Composer
2. Install [as directed](https://docs.sentry.io/platforms/php/guides/laravel/other-versions/lumen/)
3. Boot the application

### Expected Result

No errors thrown

### Actual Result

An PHP Fatal Error is raised:

```Uncaught TypeError: Sentry\\Laravel\\Http\\LaravelRequestFetcher::__construct(): Argument #1 ($app) must be of type Illuminate\\Contracts\\Foundation\\Application, Laravel\\Lumen\\Application given```

Full stack trace:

```
PHP Fatal error:  Uncaught TypeError: Sentry\\Laravel\\Http\\LaravelRequestFetcher::__construct(): Argument #1 ($app) must be of type Illuminate\\Contracts\\Foundation\\Application, Laravel\\Lumen\\Application given, called in [snip]/vendor/sentry/sentry-laravel/src/Sentry/Laravel/ServiceProvider.php on line 202 and defined in [snip]/vendor/sentry/sentry-laravel/src/Sentry/Laravel/Http/LaravelRequestFetcher.php:20
Stack trace:
#0 [snip]/vendor/sentry/sentry-laravel/src/Sentry/Laravel/ServiceProvider.php(202): Sentry\\Laravel\\Http\\LaravelRequestFetcher->__construct()
#1 [snip]/vendor/sentry/sentry/src/Integration/IntegrationRegistry.php(112): Sentry\\Laravel\\ServiceProvider->Sentry\\Laravel\\{closure}()
#2 [snip]/vendor/sentry/sentry/src/Integration/IntegrationRegistry.php(55): Sentry\\Integration\\IntegrationRegistry->getIntegrationsToSetup()
#3 [snip]/vendor/sentry/sentry/src/Client.php(107): Sentry\\Integration\\IntegrationRegistry->setupIntegrations()
#4 [snip]/vendor/sentry/sentry/src/ClientBuilder.php(165): Sentry\\Client->__construct()
#5 [snip]/vendor/sentry/sentry-laravel/src/Sentry/Laravel/ServiceProvider.php(208): Sentry\\ClientBuilder->getClient()
#6 [snip]/vendor/illuminate/container/Container.php(826): Sentry\\Laravel\\ServiceProvider->Sentry\\Laravel\\{closure}()
#7 [snip]/vendor/illuminate/container/Container.php(712): Illuminate\\Container\\Container->build()
#8 [snip]/vendor/illuminate/container/Container.php(651): Illuminate\\Container\\Container->resolve()
#9 [snip]/vendor/laravel/lumen-framework/src/Application.php(300): Illuminate\\Container\\Container->make()
#10 [snip]/vendor/laravel/lumen-framework/src/helpers.php(42): Laravel\\Lumen\\Application->make()
#11 [snip]/app/Exceptions/Handler.php(37): app()
#12 [snip]/vendor/laravel/lumen-framework/src/Concerns/RegistersExceptionHandlers.php(119): App\\Exceptions\\Handler->report()
#13 [snip]/vendor/laravel/lumen-framework/src/Concerns/RegistersExceptionHandlers.php(51): Laravel\\Lumen\\Application->handleException()
#14 [internal function]: Laravel\\Lumen\\Application->Laravel\\Lumen\\Concerns\\{closure}()
#15 {main}
  thrown in [snip]/vendor/sentry/sentry-laravel/src/Sentry/Laravel/Http/LaravelRequestFetcher.php on line 20
```

It is worth noting that this issue is only present in 2.4.0, downgrading to 2.3.1 has no problems. I suspect #419 doesn't support Lumen.

**Comment by stayallive** [comment]
I think you are absolutely right, I'm digging into it!

**Comment by stayallive** [comment]
Hey @freshleafmedia,

Would you be so kind to validate the fix from #462?

You can change your `composer.json` to this to try it out:

```diff
-   "sentry/sentry-laravel": "^2.4",
+   "sentry/sentry-laravel": "dev-fix-type-hint-laravel-lumen#64af2cb06589a76a12ac4f49b8cdc1585a3e90d3",
```

And don't forget to run `composer update` ofcourse :)

**Comment by freshleafmedia** [comment]
@stayallive Still getting the same error. I believe the fix you've written is only part of the problem. The error I am seeing I believe is coming from [src/Sentry/Laravel/Http/LaravelRequestFetcher.php:20](https://github.com/getsentry/sentry-laravel/blob/master/src/Sentry/Laravel/Http/LaravelRequestFetcher.php#L20)

**Comment by stayallive** [comment]
🤦 I should read better... how about:

```diff
-   "sentry/sentry-laravel": "^2.4",
+   "sentry/sentry-laravel": "dev-fix-type-hint-laravel-lumen#d4e05f08b81a097c6f3ccb33fdd595e56fc0699d",
```

**Comment by freshleafmedia** [comment]
All looks good :+1:

**Comment by stayallive** [comment]
Sounds great, thanks for testing with me, this will be release shortly!

**Comment by stayallive** [comment]
Fixed release [2.4.1](https://github.com/getsentry/sentry-laravel/releases/tag/2.4.1) is out 👍

**Comment by freshleafmedia** [comment]
@stayallive Thanks for the quick fix. Love open source :smiley:
